cfn-resources/util/util.go

```diff
@@ -50,6 +61,45 @@ func CreateMongoDBClient(publicKey, privateKey string) (*mongodbatlas.Client, er
 	return atlas, nil
 }
 
+func NewMongoDBClient(req handler.Request, profile *string) (*mongodbatlas.Client, *handler.ProgressEvent) {
+	profileInput := DefaultProfile
+	if profile != nil {
+		profileInput = *profile
+	}
+
+	keys, handlerError := getAPIKeys(req, profileInput)
+	if handlerError != nil {
+		return nil, handlerError
+	}
+
+	// Create atlas client
+	client, err := CreateMongoDBClient(keys.PublicKey, keys.PrivateKey)
+	if err != nil {
+		_, _ = logger.Warnf("Create - error: %+v", err)
+		peErr := progress_events.GetFailedEventByCode(fmt.Sprintf("Error creating mongoDB client : %s", err.Error()),
+			cloudformation.HandlerErrorCodeInvalidRequest)
+		return nil, &peErr
+	}
+
+	return client, nil
+}
+
+func getAPIKeys(req handler.Request, profile string) (*DeploymentSecret, *handler.ProgressEvent) {
+	key, err := GetAPIKeyFromDeploymentSecret(&req, fmt.Sprintf(ProfileName, profile))
+	if err != nil {
+		_, _ = logger.Warnf("Read - error: %+v", err)
+		pe := handler.ProgressEvent{
+			OperationStatus: handler.Failed,
+			Message: fmt.Sprintf("Error getting API-key, API-key needs to be provided using an AWS secret,"+
+				"  Ensure a secret named 'cfn/atlas/profile/%s' is created with the 'PublicKey' and 'PrivateKey' properties, error: %s",
+				profile, err.Error()),
+			HandlerErrorCode: cloudformation.HandlerErrorCodeNotFound}
+		return nil, &pe
+	}
+
+	return &key, nil
+}
+
```

Comment:
Discuss this in the standup: Could the AWS Secret manager profile stores the ProjectID and OrgID as we are currently doing with the AtlasCLI? in this way users won't need to provide this info all the time they add a new template

Reply:
+1 here, makes the profile easier to validate & is useful for future resources

Reply:
that is a good idea, i have a few concerns only, the user may have multiple projects in a single org (example 3 proj for a single org), if the user have single org level apikeys, he will need to define multiple profiles, with the same apikeys, but diferent projects, also me might need to modify, the deploy scripts, because, the deploy scripts creates a new project for each deploy in each region, once we create a project, we need to create the secret for that profile.

Reply:
> i have a few concerns only, the user may have multiple projects in a single org (example 3 proj for a single org), if the user have single org level apikeys, he will need to define multiple profiles, with the same apikeys, but diferent projects, 

This is expected. We have the same behaviour in the CLI where the user has to create multiple profiles. If the user wants to define only one profile for all the projects inside an org, they have to create Programmatic API keys at the organization level.

Reply:
Could we keep the projectId as an identifier and default to the projectId in the profile if not provided? In a profile with API keys at the org level, if the user does not want to use the default project defined in the profile, they can provide the projecId in the template.